src/file_layer.ts

```diff
@@ -37,11 +40,31 @@ type EMSFormats = EmsFileLayerFormatGeoJson | EmsFileLayerFormatTopoJson;
 export class FileLayer extends AbstractEmsService {
   protected readonly _config: FileLayerConfig;
 
+  private _getVectorDataOfType = _.memoize(
+    async (format: EMSFormatTypeStrings): Promise<FeatureCollection | TopoJSON | undefined> => {
+      const fileUrl = this.getFormatOfTypeUrl(format);
+      if (fileUrl) {
+        const vectorJson = await this._emsClient.getJsonEndpoint<
+          FeatureCollection | TopoJSON | undefined
+        >(fileUrl);
+        return vectorJson;
+      } else {
+        return;
+      }
+    }
+  );
+
   constructor(config: FileLayerConfig, emsClient: EMSClient, proxyPath: string) {
     super(config, emsClient, proxyPath);
     this._config = config;
   }
 
+  async getVectorDataOfType(
```

Comment:
thoughts on changing this to `FileLayer#getGeoJson` and doing any format conversion automagically in here if necessary?

If the default format is topojson, ems-client still downloads topojson (because it is smaller over the wire), but performs the geojson conversion "automagically". 

The reason for this is that if the conversion does not happen, the client (in this case Kibana), would still need to run the topojson->geojson conversion each time still. Conversion-cost is not _that_ huge of a penalty, but nonetheless it is one. And if ems-client already introduces caching, it may as well cache the usable converted geojson result.

The reason I think we want to avoid making this cache format-dependent is how the `suggestEMSTermJoin`-function proposed here https://github.com/elastic/kibana/pull/94969 might end up being used. 

For example, imagine a client looping over all columns in an index-pattern and calling this function with a few sample-values each to "find" a joinable layer. Even though ems-client transparently caches the downloaded result (great!), the topojson->geojson conversion would still need to happen at each call, which in a tight loop will start to be noticeable.

Reply:
+1 I like the idea of fully hiding the format logic behind the client, so it only outputs geojson features up. That way library users don't need to know or deal with TopoJSON and just consume a more familiar data representation.